src/entity/trade/TradeRecipe.php

```diff
@@ -0,0 +1,156 @@
+		private readonly ?Item $buyB = null,
+		private readonly int $maxUses = 16,
+		private readonly float $priceMultiplier = 0.05,
+		private readonly int $rewardExp = 0,
+		private int $tier = 0,
+		private int $traderExp = 0,
+		private int $uses = 0
+	){
+		if($buyA->isNull() || $sell->isNull() || ($buyB !== null && $buyB->isNull())){
+			throw new \InvalidArgumentException("Recipe items cannot be null");
+		}
+		if($tier < 0){
+			throw new \InvalidArgumentException("Tier must be a positive integer");
+		}
+	}
+
+	public function getBuyA() : Item{
+		return $this->buyA;
+	}
+
+	public function getBuyB() : ?Item{
+		return $this->buyB;
+	}
+
+	public function getSell() : Item{
+		return $this->sell;
+	}
+
+	public function getMaxUses() : int{
+		return $this->maxUses;
+	}
+
+	public function getPriceMultiplier() : float{
+		return $this->priceMultiplier;
+	}
+
+	public function getRewardExp() : int{
+		return $this->rewardExp;
+	}
+
+	public function getTier() : int{
+		return $this->tier;
+	}
+
+	public function getTraderExp() : int{
+		return $this->traderExp;
+	}
+
+	public function getUses() : int{
+		return $this->uses;
+	}
+
+	public function setUses(int $uses) : void{
+		$this->uses = $uses;
+	}
+
+	public function setTier(int $tier) : void{
+		$this->tier = $tier;
+	}
+
+	public function setTraderExp(int $traderExp) : void{
+		$this->traderExp = $traderExp;
+	}
+
+	public function isDisabled() : bool{
+		return $this->uses >= $this->maxUses;
+	}
+
+	public function nbtSerialize() : CompoundTag{
+		$nbt = CompoundTag::create()
+			->setTag(self::TAG_BUY_A, $this->buyA->nbtSerialize())
+			->setTag(self::TAG_SELL, $this->sell->nbtSerialize())
+			->setInt(self::TAG_MAX_USES, $this->maxUses)
+			->setFloat(self::TAG_PRICE_MULTIPLIER_A, $this->priceMultiplier)
+			->setFloat(self::TAG_PRICE_MULTIPLIER_B, 0.0)
+			->setInt(self::TAG_REWARD_EXP, $this->rewardExp)
+			->setInt(self::TAG_TIER, $this->tier)
+			->setInt(self::TAG_TRADER_EXP, $this->traderExp)
+			->setInt(self::TAG_USES, $this->uses);
+		if($this->buyB !== null){
+			$nbt->setTag(self::TAG_BUY_B, $this->buyB->nbtSerialize());
+		}
+		return $nbt;
+	}
+
+	public static function nbtDeserialize(CompoundTag $nbt) : self{
+		$buyATag = $nbt->getCompoundTag(self::TAG_BUY_A);
+		$sellTag = $nbt->getCompoundTag(self::TAG_SELL);
+		if($buyATag === null || $sellTag === null){
+			throw new \InvalidArgumentException("Missing buyA or sell tag in trade recipe");
+		}
+		$buyA = Item::nbtDeserialize($buyATag);
+		$sell = Item::nbtDeserialize($sellTag);
+		$buyB = null;
+		if(($buyBTag = $nbt->getCompoundTag(self::TAG_BUY_B)) !== null){
+			$buyB = Item::nbtDeserialize($buyBTag);
+		}
+		return new self(
+			$buyA,
+			$sell,
+			$buyB,
+			$nbt->getInt(self::TAG_MAX_USES, 16),
+			$nbt->getFloat(self::TAG_PRICE_MULTIPLIER_A, 0.05),
+			$nbt->getInt(self::TAG_REWARD_EXP, 0),
+			$nbt->getInt(self::TAG_TIER, 0),
+			$nbt->getInt(self::TAG_TRADER_EXP, 0),
+			$nbt->getInt(self::TAG_USES, 0)
```

Comment:
These default values are used several times, putting them in a constant would be fine